src/main/java/seedu/budgetbuddy/ExpenseList.java

```diff
@@ -219,4 +229,103 @@ public void setBudget(String category, double budget){
         budgets.add(new Budget(category, budget));
     }
 
+    /**
+     * Calculates and prints a distribution of expenses in various categories as a horizontal bar graph.
+     * It also identifies and prints the categories with the highest and lowest expenses,
+     * as well as categories where no expenses have been added.
+     */
+    public void getExpenseInsights() {
+        double totalExpenses = calculateTotalExpenses();
+        if (totalExpenses == 0) {
+            System.out.println("No expenses to display.");
+            return;
+        }
+
+        Map<String, Double> sumsByCategory = new HashMap<>();
+        for (Expense expense : expenses) {
+            sumsByCategory.merge(expense.getCategory(), expense.getAmount(), Double::sum);
+        }
+
+        // Calculate the highest expense amount
+        double highestExpense = Collections.max(sumsByCategory.values());
+
+        // Identify the categories with the highest expenses
+        List<String> highestCategories = sumsByCategory.entrySet().stream()
+                .filter(entry -> entry.getValue().equals(highestExpense))
+                .map(Map.Entry::getKey)
+                .collect(Collectors.toList());
+
+        // Calculate the lowest expense amount excluding the highest if it's the only value
+        double lowestExpense = sumsByCategory.entrySet().stream()
+                .filter(entry -> entry.getValue() > 0 && !highestCategories.contains(entry.getKey()))
+                .mapToDouble(Map.Entry::getValue)
+                .min().orElse(Double.MAX_VALUE);
+
+        // Identify the categories with the lowest expenses
+        List<String> lowestCategories = sumsByCategory.entrySet().stream()
+                .filter(entry -> entry.getValue() > 0 && entry.getValue().equals(lowestExpense))
+                .map(Map.Entry::getKey)
+                .collect(Collectors.toList());
+
+        // If lowestExpense is Double.MAX_VALUE, then this list should be empty
+        if (lowestExpense == Double.MAX_VALUE) {
+            lowestCategories.clear();
+        }
+
+        // Identify categories with no expenses
+        List<String> noExpenseCategories = categories.stream()
+                .filter(category -> !sumsByCategory.containsKey(category) || sumsByCategory.get(category) == 0)
+                .collect(Collectors.toList());
+
+        ui.printDivider();
+        printExpensesDistribution(sumsByCategory, totalExpenses);
+        ui.printDivider();
+
+        System.out.println("Highest Expense Category: " + formatCategoryList(highestCategories));
+        System.out.println("Lowest Expense Category: " + formatCategoryList(lowestCategories));
+        System.out.println("Categories with no expenses added: " + formatCategoryList(noExpenseCategories));
+        ui.printDivider();
```

Comment:
This functions seems long (About 50 LOC). Will it be better to split it into separate methods?